File: 01x10/c-010/main.rb
def theCoolerIsPrime number, primelist
    limitcheck = number
    for i in primelist do
        if i > limitcheck then
            return true
        end
        if number % i == 0 then
            return false
        else
            limitcheck = number / i
        end
    end
    true
end

def primeSumUpto limit
    primes = []
    for i in 2..limit do
        if theCoolerIsPrime(i, primes) then
            primes.push(i)
        end
    end
    primes.sum
end

puts primeSumUpto(2_000_000)

File: 01x10/c-010/main.py
def theCoolerIsPrime(number, primelist):
    limitcheck = number
    for i in primelist:
        if i > limitcheck:
            return True
        if number % i == 0:
            return False
        else:
            limitcheck = number / i
    return True

def primeSumUpto(limit):
    primes = []
    for i in range(2, limit):
        if theCoolerIsPrime(i, primes):
            primes.append(i)
    return sum(primes)

print(primeSumUpto(2_000_000))
